railties/lib/rails/generators/rails/app/templates/config/puma.rb

```diff
@@ -0,0 +1,9 @@
+threads_count = ENV.fetch("MAX_THREADS") { 5 }.to_i
```

Comment:
> If you're not likely to change these values per-environment, why do you need to be able to overwrite them ad-hoc via ENV?

A common workflow for me, *especially* regarding `MAX_THREADS`, is to tune this variable up or down 1 or 2 at a time to find the most comfortable number for the server's memory. I like that it's as easy as "change the environment variable, restart the server". 

I do the same with Puma's worker count when running in cluster mode. The memory usage of a Rails process can change after a deploy because of some code change (perhaps you added a heavy gem), and it's nice to be able to react to that with a quick "env change/restart" rather than pushing code changes through each time this variable needs to be tuned (which is fairly often on memory-constrained environments like Heroku).